### ISSUE TITLE
Allow Key Control On Clock Popups

### ISSUE THREAD

**Issue by megapickle18** [rollerderby/scoreboard#810]
Make the buttons within the clock control popups allowed to be programmed keystrokes. Simple as that.

**Comment by JeneralPain** [comment]
I'm personally struggling to understand the value here, all it does is introduce another point where someone sets up keys and forgets. Causing clocks to stop/start. Realistically if you are accurate with your start/stop jam, you should not need to adjust the clock via quick keys.

Unless there is something fundamental my personal view is missing.

**Comment by frank-weinberg** [comment]
Simple as that to describe. But not as simple under the hood:

There are not "clock control popups" (plural), there is only one clock control popup that changes clock association based on where it is opened from, causing the key assignment to suddenly control a different clock when you open the popup for a different clock. And the popup gets reset to uninitialized whenever the screen is reloaded, which includes switching games. Pressing the key in this uninitialized state would likely cause the WS handler for the screen to run into an exception and then no inputs on the screen will work until it is reloaded. So this would cause behavior that is extremely surprising for people not familiar with the internals of CRG, which is almost every user.

Aside from that: Stopping the timeout clock does NOT stop the timeout. This would not really solve the problem you're presumably trying to solve - the timeout indicator would still continue to blink, pressing a timeout type button would still change the type of the still running timeout and not start a new one and at the end of the period you'd get a "Coming Up" indicator for the period that just ended instead of Intermission/Unofficial Score. 

However, it wouldn't be a lot of effort to implement this foot gun. The effort would be in treating all the people that shoot themselves in the foot with it. If (and as long as) you (or someone else) is willing to take that burden, you can have the gun.

**Comment by megapickle18** [comment]
> There are not "clock control popups" (plural), there is only one clock control popup that changes clock association based on where it is opened from

Thanks for the explanation, that totally makes sense. Leads to the next obvious question: Can we bring back the "Show Start/Stop Buttons" button that was eliminated in v2025? When they were shown, those main-screen Start/Stop buttons were keystroke-programmable. If that's within the realm of possibility and would justify a new Feature Request thread, just say so.
 
> Stopping the timeout clock does NOT stop the timeout. 

When it's by itself, you do see how absurd this sentence is, right? :-) How 'bout a Feature Request asking that stopping the timeout clock DOES stop the timeout?

> However, it wouldn't be a lot of effort to implement this foot gun. The effort would be in treating all the people that shoot themselves in the foot with it. If (and as long as) you (or someone else) is willing to take that burden, you can have the gun.

Yes please. And thank you for speaking to the simplicity of implementing what we're asking for here.

**Comment by bullseye555** [comment]
@megapickle18 - before speedy makes this change, just confirming your purpose is to work around the current "timeout continues when jam is started" as raised in #802 
If so - I don't think this change should be made, as a better solution is being discussed in that thread.

**Comment by megapickle18** [comment]
Yes, of course.

**Comment by frank-weinberg** [comment]
> Can we bring back the "Show Start/Stop Buttons" button

It was removed because even with the buttons hidden by default too many inexperienced people managed to use them as a foot-gun. and they have no use-case outside of development that I am aware of. A Feature Request would have to make a compelling case for a use case that outweighs this foot-gun damage. 

>> Stopping the timeout clock does NOT stop the timeout.

> When it's by itself, you do see how absurd this sentence is, right? 

Periods regularly continue after the period clock has run down and stopped.
Intermissions regularly continue after the intermission clock has run down and stopped.
With the upcoming rule change lineups will regularly continue after the lineup clock has stopped (because the new rules will declare timeouts and intermissions to be part of lineup).
Jams occasionally continue after the jam clock has run down and stopped (officiating error by the JT but correct handling of that error on the scoreboard).

A period clock never continues to run after the end of the period.
An intermission clock never continues to run after the end of the intermission.
A jam clock continuing to run after the end of a jam would be an error by the SBO.
A lineup clock continuing to run after the end of lineup would be an error by the SBO.

In this context, the timeout clock behaving the same way is not absurd at all. Having it behave the exact opposite on the other hand...

> How 'bout a Feature Request asking that stopping the timeout clock DOES stop the timeout?

That's nontrivial special case handling and unnecessary - there already is a button to stop the timeout directly.

> Yes please.

Even if #812 is implemented?

**Comment by megapickle18** [comment]
> > Yes please.
> 
> Even if #812 is implemented?

No. By all means, please do that instead and put this all to rest.
